src/main/java/echobot/command/CommandParser.java

```diff
@@ -0,0 +1,128 @@
+package echobot.command;
+
+import echobot.exception.EchoBotException;
+import echobot.exception.InvalidDeadlineFormatException;
+import echobot.exception.TaskNameEmptyException;
+import echobot.exception.TaskNotFoundException;
+import echobot.exception.UnknownCommandException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommandParser {
+    public Command parse(String input) throws EchoBotException {
+        input = input.trim();
+        Pattern pattern = Pattern.compile("(?<command>\\S+)(?<arguments>.*)");
+        Matcher commandMatcher = pattern.matcher(input);
+
+        if (!commandMatcher.find()) {
+            throw new UnknownCommandException();
+        }
+        String command = commandMatcher.group("command");
+        String arguments = "";
+        if (!ListCommand.COMMAND.equals(input)) {
+            arguments = commandMatcher.group("arguments").trim();
+        }
+
+        return switch (command) {
```

Comment:
Great use of a switch block here as it really helps with the clarity of the return value for each of the cases!